### PR TITLE
Req Console Upgrade

### PR DESCRIPTION
**Pull request by SiegeB0t**
## About The Pull Request
Currently, req consoles throw a tgui error when opened by ghosts. Now ghosts can view the consoles.
## Why It's Good For The Game
I want to rage in dchat whenever a bald RO buys 500 points worth of plasteel.

## Changelog
:cl:
qol: Ghosts can view supply console information.
/:cl:

### CODE REVIEW

**Comment by FIXTHERADBOXPLEASE** [comment]
Secretly a salt PR against shipside xenos sabotaging req.

**Comment by Lumipharon** [comment]
Yeah atomise the pr. The circuit board is a balance change unrelated to ghost qol.

**Comment by SiegeB0t** [comment]
This is not a balance change. Currently, when a xeno destroys req consoles, they can be repaired with a blowtorch. However the consoles are not repairable in the case of ff. This does not make much sense. Thus, this pr is still a fix. I'll split the pr anyway.

**Comment by FIXTHERADBOXPLEASE** [comment]
Fuck I actually didn't know you could fix them with a blowtorch and I main RO. That's kind of stupid you can't fix FFed consoled then. My bad.

File: code/modules/reqs/supply.dm [hunk 1]
```diff
@@ -282,7 +282,7 @@ GLOBAL_LIST_INIT(blacklisted_cargo_types, typecacheof(list(
 	. = ..()
 	if(.)
 		return
-	if(!allowed(user))
+	if(isliving(user) && !allowed(user))
 		return
```
Comment (TiviPlus) [inline code/modules/reqs/supply.dm:286]: is this supposed to be a negation?

**Review by TiviPlus — CHANGES_REQUESTED** [review]

File: code/modules/reqs/supply.dm (hunk 1 above)
Comment (SiegeB0t) [reply code/modules/reqs/supply.dm:286]: This is correct. The function should not fail when the user is dead. allowed() works on id cards so only needs to be called if the user is living.

I could write this but it seems unnecessary with the nesting:

if(isliving(user))
	if(!allowed(user))
		return

**Review by Lumipharon — APPROVED** [review]

**Pull request merged**